src/plugins/home/public/services/add_data/add_data_service.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+/** @public */
+export interface AddDataTab {
+  id: string;
+  name: string;
+  component: React.FC;
+}
+
+export class AddDataService {
+  private addDataTabs: {
+    [key: string]: AddDataTab;
+  } = {};
```

Comment:
NIT: `Record<string, AddDataTab>`

Reply:
Updated in https://github.com/elastic/kibana/pull/100863/commits/b532118bd2a9a35510b3a5ca1389498de89f1fdd